app/app.ts

```diff
@@ -0,0 +1,156 @@
+
+import { shouldLogSession } from "../config";
+import webpackDevConfig from "../webpack/webpack.dev.js";
+
+import {
+  configureApiAuth,
+  configureInternalization,
+  configureLogger,
+  configureNunjucks,
+} from "./app-config";
+import { setupCsrfToken } from "./middleware/csrf-middleware";
+import {
+  pageNotFoundHandler,
+  serverErrorHandler,
+} from "./handlers/error-handlers";
+import {
+  logErrorMiddleware,
+  logRequestMiddleware,
+} from "./middleware/logger-middleware";
+import { router } from "./routes";
+import setupSession from "./session";
+import {
+  logSession,
+  sessionTimeOutDialogStartCheck,
+} from "./middleware/session-middleware";
+import { filterRequest } from "./middleware/xss-middleware";
+import helmet from "helmet";
+import nocache from "nocache";
+import { fetchCookie } from "./middleware/cookie-middleware";
+import { existsSync } from "fs";
+import { setLocalVars } from "./middleware/set-locals";
+import useragent from "express-useragent";
+import { pathName } from "./paths";
+
+const crypto = require("crypto");
+const fs = require("fs");
+
+const fileHash = (filename: string): string => {
+  if (existsSync(filename)) {
+    const file_buffer = fs.readFileSync(filename);
+    const sum = crypto.createHash("md5");
+    sum.update(file_buffer);
+    const hex = sum.digest("hex");
+    return hex;
+  } else {
+    return "dev";
+  }
+};
+
+const createApp = (): express.Application => {
+  const app: express.Application = express();
+  const node_env: string = process.env.NODE_ENV;
+  app.locals.environment = process.env.ENVIRONMENT;
+  app.locals.knownPaths = [];
+  const publicKeys = Object.keys(pathName.public);
+  publicKeys.forEach((key) => {
+    const pth = pathName.public[key];
+    app.locals.knownPaths.push(pth);
+  });
+
+  app.locals.allJsSuffix = "ref=" + fileHash("./build/all.js");
+  app.locals.styleCssSuffix = "ref=" + fileHash("./build/style.css");
+  app.use(useragent.express());
+  app.use(helmet());
+  app.use(
+    helmet.contentSecurityPolicy({
+      directives: {
+        defaultSrc: ["'self'"],
+        frameSrc: ["https://www.googletagmanager.com"],
+        imgSrc: [
+          "'self'",
+          "data:",
+          "https://www.googletagmanager.com",
+          "https://www.google-analytics.com",
+        ],
+        scriptSrc: [
+          "'self'",
+          "'sha256-+6WnXIl4mbFTCARd8N3COQmT3bJJmo32N8q8ZSQAIcU='",
+          "'sha256-9XY5Gy2dQzMz7vEBVmHq5oL6lwMY5OD8GXGFHG1FIt0='",
+          "'sha256-8EAj23is6S15MqKEZyEUwkySwX+ECmNHCS48+5c9ycY='",
+          "'sha256-gGKtWVlwtb9dj1Sa7K4ybKpqUOE61nkachaCJ1LfmFY='",
+          "'sha256-mucrrw5J6PKI6zn3xs64Q7aaThRouvSD1FKsWFg24Wk='",
+          "'sha256-sNEhyAswTBSPnegdEB1z/BKKUaCXs0HeKkW28zu+7Uw='",
+          "https://www.googletagmanager.com",
+          "https://www.google-analytics.com",
+          "https://ssl.google-analytics.com",
+          "https://www.tax.service.gov.uk",
```

Comment:
Might want to remove this too

Reply:
I will remove these if we don't want google analytics
tax service is for the session timeout dialogs as they were written by HMRC

Reply:
looks like tax.service no longer used in latest version, will remove